src/Microsoft.AspNetCore.Blazor/Components/ParameterEnumerator.cs

```diff
@@ -28,9 +28,20 @@ internal ParameterEnumerator(RenderTreeFrame[] frames, int ownerIndex)
         /// Gets the current value of the enumerator.
         /// </summary>
         public Parameter Current
-            => _currentIndex > _ownerIndex
-                ? new Parameter(_frames, _currentIndex)
-                : throw new InvalidOperationException("Iteration has not yet started.");
+        {
+            get
+            {
+                if (_currentIndex > _ownerIndex)
+                {
+                    ref var frame = ref _frames[_currentIndex];
+                    return new Parameter(frame.AttributeName, frame.AttributeValue);
+                }
+                else
+                {
+                    throw new InvalidOperationException("Iteration has not yet started.");
+                }
+            }
```

Comment:
At this point I'd suggest just doing this inside `MoveNext()` and not bothering to throw an exception - the behavior of accessing an enumerator before calling `MoveNext()` isn't defined, so returning a default value is totally acceptable.